packages/envd/internal/api/store.go

```diff
@@ -2,23 +2,134 @@ package api
 
 import (
 	"encoding/json"
+	"fmt"
 	"net/http"
-
-	"github.com/rs/zerolog"
+	"slices"
 
 	"github.com/e2b-dev/infra/packages/envd/internal/host"
 	"github.com/e2b-dev/infra/packages/envd/internal/utils"
+	"github.com/e2b-dev/infra/packages/shared/pkg/keys"
+	"github.com/rs/zerolog"
+)
+
+const (
+	SigningReadOperation  = "read"
+	SigningWriteOperation = "write"
+)
+
+var (
+	accessTokenHeader = "X-Access-Token"
+
+	// paths that are always allowed without general authentication
+	allowedPaths = []string{
+		"GET/health",
+		"GET/files",
+		"POST/files",
+	}
 )
 
 type API struct {
-	logger  *zerolog.Logger
-	envVars *utils.Map[string, string]
+	logger      *zerolog.Logger
+	accessToken *string
+	envVars     *utils.Map[string, string]
+}
+
+type jsonErrorResponse struct {
+	Error string `json:"error"`
 }
 
 func New(l *zerolog.Logger, envVars *utils.Map[string, string]) *API {
 	return &API{logger: l, envVars: envVars}
 }
 
+func (a *API) WithAuthorization(handler http.Handler) http.Handler {
```

Comment:
I would probably move all these to separate file to keep it clean

Reply:
done